src/universal/modules/meeting/components/MeetingCheckInPrompt/CheckInQuestion.js

```diff
@@ -0,0 +1,261 @@
+    }
+    return undefined;
+  };
+
+  handleBeforeInput = (char) => {
+    const {handleBeforeInput} = this.props;
+    if (handleBeforeInput) {
+      return handleBeforeInput(char);
+    }
+    return undefined;
+  }
+
+  handlePastedText = (text) => {
+    if (text) {
+      for (let i = 0; i < textTags.length; i++) {
+        const tag = textTags[i];
+        if (text.indexOf(tag) !== -1) {
+          const selection = this.props.editorState.getSelection();
+          this.entityPasteStart = {
+            anchorOffset: selection.getAnchorOffset(),
+            anchorKey: selection.getAnchorKey()
+          };
+        }
+      }
+    }
+    return 'not-handled';
+  };
+
+  selectAllQuestion = () => {
+    this.editorRef.focus();
+    const {editorState, setEditorState} = this.props;
+    const selection = editorState.getSelection();
+    const contentState = editorState.getCurrentContent();
+    const fullSelection = selection.merge({
+      anchorKey: contentState.getFirstBlock().getKey(),
+      focusKey: contentState.getLastBlock().getKey(),
+      anchorOffset: 0,
+      focusOffset: contentState.getLastBlock().getLength()
+    });
+    const nextEditorState = EditorState.forceSelection(editorState, fullSelection);
+    setEditorState(nextEditorState);
+  };
+
+  render() {
+    const {editorState, canEdit, styles} = this.props;
+    const {isEditing} = this.state;
+    const tip = canEdit
+      ? 'Tap to customize'
+      : 'Upgrade to a Pro Account to customize the Social Check-in question.';
+    return (
+      <Tooltip
+        tip={<div>{tip}</div>}
+        originAnchor={{vertical: 'bottom', horizontal: 'center'}}
+        targetAnchor={{vertical: 'top', horizontal: 'center'}}
+        hideOnFocus
+      >
+        <div className={css(styles.root)}>
+          <Editor
+            blockStyleFn={this.blockStyleFn}
+            editorState={editorState}
+            handleBeforeInput={this.handleBeforeInput}
+            handleKeyCommand={this.handleKeyCommand}
+            handlePastedText={this.handlePastedText}
+            handleReturn={this.handleReturn}
+            keyBindingFn={this.keyBindingFn}
+            onChange={this.handleChange}
+            onDownArrow={this.handleDownArrow}
+            onEscape={this.handleEscape}
+            onTab={this.handleTab}
+            onUpArrow={this.handleUpArrow}
+            placeholder="e.g. How are you?"
+            readOnly={!canEdit}
+            onFocus={this.setLocalEditingState(true)}
+            onBlur={this.setLocalEditingState(false)}
+            ref={(c) => {
+              this.editorRef = c;
+            }}
+          />
+          {canEdit && !isEditing &&
+            <FontAwesome
+              role="button"
+              aria-label={tip}
+              tabIndex="0"
+              name="pencil"
+              style={{...iconStyle, ...buttonStyle}}
+              onClick={this.selectAllQuestion}
+              onKeyUp={(event) => {
```

Comment:
what's this do?

Reply:
This is for emulating a button! Since you wanted that icon to be clickable, we gotta make sure it's accessible (e.g. tells the browser "I'm a button!") by:

- adding `role="button"` ("Think of me as a button!")
- adding `tabIndex="0"` ("Insert me into the tab order, just like a button would be!")
- making the custom element clickable via the spacebar, which is what this does.

Reply:
oh i love it.
we should make this more declarative though. i'll build a component